src/channel/rpc_0_6.ts

```diff
@@ -22,11 +22,12 @@ import { CallData } from '../utils/calldata';
 import { isSierra } from '../utils/contract';
 import { validateAndParseEthAddress } from '../utils/eth';
 import fetch from '../utils/fetchPonyfill';
-import { getSelector, getSelectorFromName } from '../utils/hash';
+import { computeHashOnElements, getSelector, getSelectorFromName } from '../utils/hash';
 import { stringify } from '../utils/json';
 import { getHexStringArray, toHex, toStorageKey } from '../utils/num';
 import { Block, getDefaultNodeUrl, isV3Tx, isVersion, wait } from '../utils/provider';
 import { decompressProgram, signatureToHexArray } from '../utils/stark';
+import { useEncoded } from '../utils/starknetId';
```

Comment:
I don't think we used anything from StarknetId utils anywhere else in the code, doesn't look like we should do it here as well? cc: @tabaktoni

Reply:
Yep, not sure this is a correct encoder.

Reply:
That was the closest I found that could carry out the operation. You know of any I could use instead? @tabaktoni

Reply:
It is not only about using encode, the result of the getL1MessageHash should be **L1 message hash**.
Did you check that this produces the correct hash?

Reply:
> I think it should work since the formula is implemented correctly 🙂. Can you help with a sample to verify with? @tabaktoni I am having a hard time getting one

https://sepolia.starkscan.co/tx/0x024f985090da17ccc9734efd2f516262fce50b37c702bf1e9d2a149796c05cfd
I am using this transaction to test